### DISCUSSION TITLE
Remove, add item using useFieldArray

### DISCUSSION THREAD

**Discussion by ronaldruzicka** [react-hook-form/react-hook-form#2410] [Q&A] [answered]
Hi, I'm having problems with removing and adding fields using `useFieldArray`. I have a state from where I get data, when I fetch the data I inject the data to `defaultValues`. I reset the form using `reset` when values change, otherwise the values in the form won't get updated after data is fetched. Then I have `add` and `remove` buttons.

1. When I add new input, it doesn't have the values I add to it via `append` function
2. When I remove one input and then add a new one, it has the value from the removed input

In both cases I would expect to have the values from the `append` function, which in my case should be `0, 0`

code sandbox: https://codesandbox.io/s/empty-fast-lhgep?file=/src/Form.tsx

Any help would be appreciated!

**Comment by ronaldruzicka** [comment, marked as answer]
Ok so I figured it out. I was not passing `defaultValue` to the `Controller`. Stupid me! 🤦‍♂️

**Reply by bluebill1049** [reply]
oops sorry I completely miss your question, glad you sort it out 🙏
